lighthouse-core/audits/autocomplete.js

```diff
@@ -43,30 +185,83 @@ class AutocompleteAudit extends Audit {
     };
   }
 
+  /**
+   * @param {LH.Artifacts.FormInput} input
+   * @return {{isValid: Boolean, orderWarn?: Boolean}}
+   */
+  static isValidAutocomplete(input) {
+    if (!input.autocomplete.attribute) return {isValid: false};
+    if (input.autocomplete.attribute.includes(' ') ) {
+      const tokenArray = input.autocomplete.attribute.split(' ');
+      for (const token of tokenArray) {
+        if (token.slice(0, 8) === 'section-') continue;
+        if (validAutocompleteTokens.includes(token)) continue;
+        return {isValid: false};
+      }
+      if (!input.autocomplete.property) return {isValid: false, orderWarn: true};
+      return {isValid: true};
+    }
+    return {isValid: validAutocompleteTokens.includes(input.autocomplete.attribute)};
+  }
+
   /**
    * @param {LH.Artifacts} artifacts
    * @return {LH.Audit.Product}
    */
   static audit(artifacts) {
     const forms = artifacts.FormElements;
     const failingFormsData = [];
+    const warnings = [];
+    let notApplicable = 0;
+    let inputsCount = 0;
     for (const form of forms) {
       for (const input of form.inputs) {
-        if (!input.autocomplete) {
-          failingFormsData.push({
-            node: /** @type {LH.Audit.Details.NodeValue} */ ({
-              type: 'node',
-              snippet: input.snippet,
-              nodeLabel: input.nodeLabel,
-            }),
-          });
+        inputsCount += 1;
+        const token = this.isValidAutocomplete(input);
+        if (!token.isValid) {
```

Comment:
currently this method doesn't handle the case of where the tokens are good, but the prediction doesn't match. let's add?

Reply:
further more.. i think if you write a little `doesPredictionMatch` helper function, that'll help structure this logic.

Reply:
this is for cases where the required autocomplete token is valid but the optional tokens are not?

Reply:
@paulirish

Reply:
doesPredictionMatch is that the prediction is already one of the valid autocomplete tokens.

maybe it'd be better as doesPredictionDiffer tbh....  not sure

Reply:
![Screen Shot 2020-08-31 at 9 07 40 PM](https://user-images.githubusercontent.com/34068912/91783364-04877d00-ebce-11ea-8e0a-49ed716686ec.png)

for cases like this?

Reply:
in this case the prediction is actually more accurate though. street-adress is valid but if you look at the input all it need is address-line1 not the whole street address